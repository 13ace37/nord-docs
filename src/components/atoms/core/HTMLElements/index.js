/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

/**
 * @file Provides components that represents basic HTML elements.
 * @author Arctic Ice Studio <development@arcticicestudio.com>
 * @author Sven Greb <development@svengreb.de>
 * @see https://developer.mozilla.org/en-US/docs/Web/HTML/Element
 * @since 0.3.0
 */

import { A } from "./inlineTextSemantics";

/* eslint-disable-next-line import/prefer-default-export */
export { A };
