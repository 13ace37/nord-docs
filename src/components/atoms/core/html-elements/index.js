/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

/**
 * @file Provides components that represent basic HTML elements.
 * @author Arctic Ice Studio <development@arcticicestudio.com>
 * @author Sven Greb <development@svengreb.de>
 * @since 0.3.0
 * @see https://developer.mozilla.org/en-US/docs/Web/HTML/Element
 */

import Input from "./forms";
import { A, Code } from "./inline-text-semantics";
import { Details, Summary } from "./interactive";
import { H1, H2, H3, H4, H5, H6 } from "./sectioning";
import { Caption, Table, Td, Th, Tr } from "./tabular";
import { Blockquote, Figure, Hr, Li, Ol, P, Pre, Ul } from "./text";

export {
  A,
  Blockquote,
  Caption,
  Code,
  Details,
  Figure,
  H1,
  H2,
  H3,
  H4,
  H5,
  H6,
  Hr,
  Input,
  Li,
  Ol,
  P,
  Pre,
  Summary,
  Table,
  Td,
  Th,
  Tr,
  Ul
};
