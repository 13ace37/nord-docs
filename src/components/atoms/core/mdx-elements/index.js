/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import Image from "./Image";
import ShrinkedWidth from "./ShrinkedWidth";
import Video from "./Video";

export { Image, ShrinkedWidth, Video };
