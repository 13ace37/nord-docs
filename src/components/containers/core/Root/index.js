/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import Root, { GlobalThemeMode } from "./Root";

export { GlobalThemeMode };
export default Root;
