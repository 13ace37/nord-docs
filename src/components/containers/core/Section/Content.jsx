/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import styled from "styled-components";

import CoreContent from "containers/core/Content";

/**
 * A container for content of the `Section` component.
 *
 * @author Arctic Ice Studio <development@arcticicestudio.com>
 * @author Sven Greb <development@svengreb.de>
 * @since 0.3.0
 */
const Content = styled(CoreContent)`
  padding-top: ${({ compact }) => !compact && "5em"};
  padding-bottom: ${({ compact }) => !compact && "5em"};
`;

export default Content;
