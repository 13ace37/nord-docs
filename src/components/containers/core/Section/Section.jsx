/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import PropTypes from "prop-types";
import styled from "styled-components";

import { motion } from "styles/theme";

import {
  baseBackgroundColor,
  primaryBackgroundColor,
  secondaryBackgroundColor,
  tertiaryBackgroundColor
} from "../shared/styles";

const variants = {
  base: baseBackgroundColor,
  primary: primaryBackgroundColor,
  secondary: secondaryBackgroundColor,
  tertiary: tertiaryBackgroundColor
};

/**
 * A base HTML component that represents a `<section>` with multiple base style variants.
 *
 * @author Arctic Ice Studio <development@arcticicestudio.com>
 * @author Sven Greb <development@svengreb.de>
 * @see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/section
 * @see https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Using_HTML_sections_and_outlines
 * @since 0.3.0
 */
const Section = styled.section`
  background-color: ${({ variant }) => variants[variant]};
  transition: background-color ${motion.speed.duration.transition.base.themeModeSwitch}ms ease-in-out;
  position: ${({ decorated }) => decorated && "relative"};
`;

Section.propTypes = {
  variant: PropTypes.oneOf(Object.keys(variants))
};

Section.defaultProps = {
  variant: "base"
};

export default Section;
