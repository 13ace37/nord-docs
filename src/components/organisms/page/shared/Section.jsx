/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import styled from "styled-components";

import CoreSection from "containers/core/Section";

/**
 * A main container that applies section specific styles and CSS workarounds.
 *
 * @author Arctic Ice Studio <development@arcticicestudio.com>
 * @author Sven Greb <development@svengreb.de>
 * @since 0.6.0
 */
const Section = styled(CoreSection)`
  /* The relative position is necessary for sections that render absolute positioned decorations. */
  position: relative;
  /*
   * Due to a bug in Firefox there is a one pixel gap between the SVG section divider (skewed wave) of the previous
   * section and this section. The workaround moves this section up by the amount of pixels specified by the "gapWidth"
   * prop to ensure the gap is closed. Values larger than one pixel are necessary because the previous section is
   * affected by the same problem and applied the same workaround which results in a gap larger than one pixel between
   * these sections.
   */
  top: ${({ gapWidth }) => gapWidth && `-${gapWidth}px`};
  /* Some sections need to use special background colors to differ from the previous and/or next section. */
  background-color: ${({ bgColor }) => bgColor};
`;

export default Section;
