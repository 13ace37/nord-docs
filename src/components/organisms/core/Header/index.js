/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import { HEADER_BRAND_LOGO_SHRINK_FACTOR, HEADER_HEIGHT, HEADER_HEIGHT_PINNED } from "./shared/styles";
import Header from "./Header";

export { HEADER_BRAND_LOGO_SHRINK_FACTOR, HEADER_HEIGHT, HEADER_HEIGHT_PINNED };
export default Header;
