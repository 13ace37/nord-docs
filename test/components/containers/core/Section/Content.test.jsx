/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import React from "react";
import { stripUnit } from "polished";

import { Content } from "containers/core/Section";
import { renderWithTheme } from "nord-docs-test-utils";

describe("theme styles", () => {
  test("has top and bottom padding", () => {
    const { container } = renderWithTheme(<Content>Nord</Content>);
    const paddingTop = stripUnit(getComputedStyle(container.firstChild).paddingTop);
    const paddingBottom = stripUnit(getComputedStyle(container.firstChild).paddingBottom);

    expect(paddingTop).toBeGreaterThan(0);
    expect(paddingBottom).toBeGreaterThan(0);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("adjusts top and bottom padding to passed props", () => {
    const { container } = renderWithTheme(<Content compact>Nord</Content>);
    const paddingTop = stripUnit(getComputedStyle(container.firstChild).paddingTop);
    const paddingBottom = stripUnit(getComputedStyle(container.firstChild).paddingBottom);

    expect(paddingTop).toBeFalsy();
    expect(paddingBottom).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
