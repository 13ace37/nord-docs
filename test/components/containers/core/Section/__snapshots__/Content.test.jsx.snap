// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`theme styles adjusts top and bottom padding to passed props 1`] = `
.c0 {
  width: 100%;
  padding: 0 1.5em;
}

@media (min-width:75em) {
  .c0 {
    max-width: 75em;
    padding: unset;
  }
}

<div
  class="c0"
>
  Nord
</div>
`;

exports[`theme styles has top and bottom padding 1`] = `
.c0 {
  width: 100%;
  padding: 0 1.5em;
  padding-top: 2.5em;
  padding-bottom: 2.5em;
}

@media (min-width:75em) {
  .c0 {
    max-width: 75em;
    padding: unset;
  }
}

@media (min-width:56.25em) {
  .c0 {
    padding-top: 5em;
    padding-bottom: 5em;
  }
}

<div
  class="c0"
>
  Nord
</div>
`;
